File: 1979-find-greatest-common-divisor-of-array/1979-find-greatest-common-divisor-of-array.py
class Solution:
    def findGCD(self, nums: List[int]) -> int:
        nums.sort()
        a=nums[0]
        b=nums[-1]
        return self.c(a,b)
    def c(self,a:int,b:int)-> int:
        while b!=0:
            a,b=b,a%b
        return abs(a)
           
        
        
        

File: 1979-find-greatest-common-divisor-of-array/1979-find-greatest-common-divisor-of-array.java
class Solution {
    public int findGCD(int[] nums) {
        Arrays.sort(nums);
        int num1=nums[0];
        int num2=nums[nums.length-1];      
        return gets(num1,num2);
    }
        public static int gets(int a,int b){
             while(b!=0){
               int temp=b;
               b=a%b;
               a=temp;
             } 
        return Math.abs(a);
   }
}